### PR TITLE
1173 Add dump env helper

### PR DESCRIPTION
**Pull request by Wauplin**
Resolve https://github.com/huggingface/huggingface_hub/issues/1173. Should ease some debugging of issues from users.

TODO:
- [x] update issue template ?
- [x] is there any other relevant information to paste ?

Inspired by existing work in `transformers` `diffusers` (thanks @mariosasko)

### In terminal

```
➜  huggingface-cli env

Copy-and-paste the text below in your GitHub issue.

- huggingface_hub version: 0.11.0.dev0
- Platform: Linux-5.15.0-52-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- Running in iPython ?: No
- Running in notebook ?: False
- Running in Google Colab ?: False
- Token path ?: /home/wauplin/.huggingface/token
- Has saved token ?: True
- Token type: Personal token
- Who am I ?: Wauplin
- Configured git credential helpers: 
- FastAI: N/A
- Tensorflow: 2.9.1
- Torch: 1.12.1
- Jinja2: 3.1.2
- Graphviz: 0.20.1
- Pydot: 1.4.2
```

### From python

```py
from huggingface_hub import dump_environment_info

dump_environment_info()
```

Output from ipython in terminal:
```text

Copy-and-paste the text below in your GitHub issue.

- huggingface_hub version: 0.11.0.dev0
- Platform: Linux-5.15.0-52-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- Running in iPython ?: Yes
- iPython shell: TerminalInteractiveShell
- Running in notebook ?: False
- Running in Google Colab ?: False
- Token path ?: /home/wauplin/.huggingface/token
- Has saved token ?: True
- Token type: Personal token
- Who am I ?: Wauplin
- Configured git credential helpers: 
- FastAI: N/A
- Tensorflow: 2.9.1
- Torch: 1.12.1
- Jinja2: 3.1.2
- Graphviz: 0.20.1
- Pydot: 1.4.2
```

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM! You could also edit the issue templates to request the command to be run before opening an issue. Right now it asks for the versions, but it would be simpler to just point to the command

File: src/huggingface_hub/utils/_runtime.py [hunk 1]
```diff
@@ -177,3 +179,56 @@ def is_google_colab() -> bool:
...
+        info["Running in iPython ?"] = "No"
+    info["Running in notebook ?"] = is_notebook()
+    info["Running in Google Colab ?"] = is_google_colab()
+
+    # Login info
+    info["Token path ?"] = HfFolder().path_token
+    info["Has saved token ?"] = token is not None
+    if token is not None:
+        info["Token type"] = (
+            "Organization token" if token.startswith("hf_org") else "Personal token"
```
Comment (osanseviero) [inline src/huggingface_hub/utils/_runtime.py:212]: I don't think we can ever have an organization token saved right? The login fails for those (see https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/_login.py#L225 )

**Review by osanseviero — APPROVED** [review]
Thanks! Agree on changing the templates

File: src/huggingface_hub/utils/_runtime.py (hunk 1 above)
Comment (Wauplin) [reply src/huggingface_hub/utils/_runtime.py:212]: Yep true, I'll remove it. Thanks :)

**Comment by Wauplin** [comment]
Thanks both for the review ! 

>  You could also edit the issue templates to request the command to be run before opening an issue.

Yes that's what I meant by " update issue template ?" in the todo list of the PR but it wasn't so clear :smile: 

Anyway, I updated it and will merge when CI is green. Hope it will prove to be useful in the future :)

File: src/huggingface_hub/utils/_runtime.py (hunk 1 above)
Comment (julien-c) [reply src/huggingface_hub/utils/_runtime.py]: yes i was going to comment the same. Org tokens are on their way to deprecation anyways

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -26,12 +26,31 @@ body:
...
+    attributes:
+      value: |
+        ## System Info
+
+        Please dump your environment info by running the following command and copy-paste the result here:
+        ```txt
+        huggingface-cli env
+        ```
+
+        If your are working in a notebook, please run it in a code cell:
```
Comment (julien-c) [inline .github/ISSUE_TEMPLATE/bug-report.yml:39]: *you

File: src/huggingface_hub/utils/_runtime.py [hunk 3]
```diff
@@ -177,3 +179,53 @@ def is_google_colab() -> bool:
     Taken from https://stackoverflow.com/a/63519730.
     """
     return _is_google_colab
+
+
+def dump_environment_info() -> Dict[str, Any]:
```
Comment (julien-c) [inline src/huggingface_hub/utils/_runtime.py:184]: (nit) maybe link to `transformers` `diffusers` implementations for reference

**Review by julien-c — COMMENTED** [review]
lgtm!

**Pull request merged**